profiler/src/ProfilerEngine/Datadog.Profiler.Native/StackSnapshotResultBuffer.h

```diff
@@ -179,27 +174,30 @@ inline void StackSnapshotResultBuffer::DetermineAppDomain(ThreadID threadId, ICo
 
 inline bool StackSnapshotResultBuffer::AddFrame(std::uintptr_t ip)
 {
-    if (_currentFramesCount >= MaxSnapshotStackDepth_Limit)
-    {
-        return false;
-    }
-
-    if (_currentFramesCount == MaxSnapshotStackDepth_Limit - 1)
-    {
-        _instructionPointers[_currentFramesCount++] = 0;
-        return false;
-    }
-
-    _instructionPointers[_currentFramesCount++] = ip;
-    return true;
+    return _callstack.Add(ip);
 }
 
 inline bool StackSnapshotResultBuffer::AddFakeFrame()
 {
     return AddFrame(0);
 }
 
-inline uintptr_t* StackSnapshotResultBuffer::Data()
+inline shared::span<uintptr_t> StackSnapshotResultBuffer::Data()
+{
+    return _callstack.Data();
+}
+
+inline Callstack StackSnapshotResultBuffer::GetCallstack()
```

Comment:
Why do we need this method that "move" the callstack in addition to Data()?
Could having both lead to ownership bugs?

Reply:
`Data()` does not move : it just offers a view (span) on the callstack. It's the same as the previous implementation.
`Data() + SetCount` are used by Linux
`Add()` API is used by Windows